modules/apps/transfer/keeper/relay_forwarding_test.go

```diff
@@ -572,13 +572,10 @@ func (suite *KeeperTestSuite) TestAcknowledgementFailureScenario5Forwarding() {
 	err = suite.chainB.GetSimApp().TransferKeeper.OnAcknowledgementPacket(suite.chainB.GetContext(), packetRecv, data, ack)
 	suite.Require().NoError(err)
 
-	// Check that Escrow B has been refunded amount
-	// NOTE This is failing. The revertInFlightsChanges sohuld mint back voucher to chainBescrow
-	// but this is not happening. It may be a problem related with how we're writing async acks.
-	//
+	// Check that Escrow B is empty the revertInFlightsChanges should burn the entire token since it is not native to Chain B.
```

Comment:
Mmm, I think that at this point amount should have been refunded to Escrow B.

We start the test with 2 simple transfers.

A -> B
B -> C
At the end of this two transfer we verify at lines 460-470 that states are:

EscrowB = Amount (denom AB)
UserC = Amount (denom ABC)
The states we see here are the ones we should expect to see at the end of the test.
Indeed then we start the forwarding transfer C -> B -> A and we force this transfer to fail.
So we should expect that all the states are reverted back to the point before the forwarding transfer happen. Which means we should go back just right after the simple transfer B->C has finalised. So final states should be:

EscrowB = Amount (denom AB)
UserC = Amount (denom ABC)